lumicks/pylake/population/dwelltime.py

```diff
@@ -145,37 +149,52 @@ def plot_axes(data, key, component, use_index):
         plt.tight_layout()
 
 
-@dataclass(frozen=True)
-class BindingDwelltimes:
-    """Results of exponential mixture model optimization for binding dwelltimes.
+class DwelltimeModel:
+    """Exponential mixture model optimization for dwelltime analysis.
 
-    This class is returned from `_kinetic_mle_optimize()` and should not be
-    constructed manually.
-
-    Attributes
+    Parameters
     ----------
-    n_components : int
-        number of components in the model.
     dwelltimes_sec : np.ndarray
         observations on which the model was trained.
-    observations_limits : tuple
-        tuple of (`min`, `max`) values of the experimental observation time.
-    _parameters : np.ndarray
-        optimized parameters in the order [amplitudes, lifetimes]
-    log_likelihood : float
-        log likelihood of the trained model
-    bootstrap : BindingDwelltimesBootstrap
-        object containing information about the bootstrapping analysis.
+    n_components : int
+        number of components in the model.
+    min_observation_time : float
+        minimum experimental observation time
+    max_observation_time : float
+        maximum experimental observation time.
+    tol : float
+        The tolerance for optimization convergence.
+    max_iter : int
+        The maximum number of iterations to perform.
     """
 
-    n_components: int
-    dwelltimes_sec: np.ndarray = field(repr=False)
-    observation_limits: list = field(repr=False)
-    _parameters: np.ndarray = field(repr=False)
-    log_likelihood: float
-    bootstrap: BindingDwelltimesBootstrap = field(
-        default_factory=BindingDwelltimesBootstrap, init=False, repr=False
-    )
+    def __init__(
+        self,
+        dwelltimes_sec,
```

Comment:
Does it have to be seconds?

Reply:
I guess not. I originally named it like that to hint that if you want the lifetime in seconds, the input needs to be in units of seconds. but I'll just beef up the docstring